src/otaclient/create_standby/resume_ota.py

```diff
@@ -149,3 +190,142 @@ def resume_ota(self) -> None:
             logger.warning(f"exception during scanning OTA resource dir: {e!r}")
         finally:
             os.close(_resource_dir_fd)
+            self._internal_que.put_nowait(None)
+            status_reporter_t.join()
+
+
+class ResourceStreamer(_ResourceOperatorBase):
+    """Scan and verify OTA resource folder at the source slot, and stream with verifying it
+    to the destination slot.
+    """
+
+    def __init__(
+        self,
+        *,
+        all_resource_digests: ResourcesDigestWithSize,
+        src_resource_dir: Path,
+        dst_resource_dir: Path,
+        status_report_queue: Queue[StatusReport],
+        session_id: str,
+    ) -> None:
+        self._status_report_queue = status_report_queue
+        self.session_id = session_id
+        self._src_resource_dir = src_resource_dir
+        self._dst_resource_dir = dst_resource_dir
+        self._all_resource_digests = all_resource_digests
+
+        self._se = threading.Semaphore(cfg.MAX_CONCURRENT_PROCESS_FILE_TASKS)
+        self._thread_local = threading.local()
+        self._internal_que: Queue[int | None] = Queue()
+
+    def _process_resource_at_thread(
+        self,
+        expected_digest: bytes,
+        expected_digest_hex: str,
+        *,
+        src_dir_fd: int,
+        dst_dir_fd: int,
+    ) -> None:
+        tmp_dst_fname = _gen_tmp_fname()
+        try:
+            hash_f, file_size = sha256(), 0
+            buffer, bufferview = (
+                self._thread_local.buffer,
+                self._thread_local.bufferview,
+            )
+
+            src_fd = os.open(expected_digest_hex, os.O_RDONLY, dir_fd=src_dir_fd)
+            tmp_dst_fd = os.open(
+                tmp_dst_fname,
+                os.O_WRONLY | os.O_CREAT | os.O_TRUNC,
+                dir_fd=dst_dir_fd,
+            )
+
+            with open(src_fd, "rb") as src, open(tmp_dst_fd, "wb") as tmp_dst:
+                os.posix_fadvise(src_fd, 0, 0, os.POSIX_FADV_NOREUSE)
+                os.posix_fadvise(src_fd, 0, 0, os.POSIX_FADV_SEQUENTIAL)
+                os.posix_fadvise(tmp_dst_fd, 0, 0, os.POSIX_FADV_NOREUSE)
+                os.posix_fadvise(tmp_dst_fd, 0, 0, os.POSIX_FADV_SEQUENTIAL)
```

Comment:
in my understanding, this `POSIX_FADV_SEQUENTIAL` is the hint to read access.

Reply:
Removed in [43cbf00](https://github.com/tier4/ota-client/pull/649/commits/43cbf00f5986a5e9d339039b755e5e897fc488ab).